CMakeLists.txt

```diff
@@ -431,8 +431,31 @@ endif()
 
 if(DEFINED BUILD_VERSION)
   set(build_version_argument "-DBUILD_VERSION=${BUILD_VERSION}")
-elseif(EXISTS ${ZEPHYR_BASE}/.git/index)
+elseif(IS_DIRECTORY ${ZEPHYR_BASE}/.git)
   set(git_dependency ${ZEPHYR_BASE}/.git/index)
+elseif(EXISTS ${ZEPHYR_BASE}/.git)
+  # Likely a git-submodule. Let's ask git where the real database is located.
+  find_package(Git QUIET)
+  if(GIT_FOUND)
+    execute_process(
+      COMMAND ${GIT_EXECUTABLE} rev-parse --git-dir
+      WORKING_DIRECTORY ${ZEPHYR_BASE}
+      OUTPUT_VARIABLE KERNEL_GIT_DIR
```

Comment:
`KERNEL_GIT_DIR` is really a misnomer since the "zephyr" repository is not just kernel, but all the core components that constitute the "Zephyr RTOS."

A more generic `ZEPHYR_GIT_DIR` would be a better name.

Reply:
agreeing with this.

Also note that locally used variables should be written in lower case, that is: `zephyr_git_dir`

Reply:
Makes sense, done.